aten/src/ATen/cpu/vec256/vec256_base.h

```diff
@@ -157,7 +157,7 @@ struct Vec256 {
   static Vec256<T> arange(T base = static_cast<T>(0), step_t step = static_cast<step_t>(1)) {
     Vec256 vec;
     for (int64_t i = 0; i < size(); i++) {
-      vec.values[i] = base + i * step;
+      vec.values[i] = base + static_cast<step_t>(i) * step;
```

Comment:
what is this for?

I thought `arange` is not supported for complex dtypes?

Reply:
It is used in linspace

Reply:
hmm maybe we should add python API for `arange` for complex dtypes since numpy supports it too.

cc. @mruberry

Reply:
Is this for a complex `step`? This multiplication should have been enabled in `c10/util/Complex.h`: https://github.com/pytorch/pytorch/pull/34506/files

Reply:
OK, I see, you have switched to `c10::complex`. Do you think it is desirable to add something similar to `c10/util/Complex.h` for `c10::complex`? This could make code look cleaner and also improve efficiency.